tesseract_kinematics/include/tesseract_kinematics/ikfast/ikfast_inv_kin.h

```diff
@@ -160,6 +162,7 @@ class IKFastInvKin : public InverseKinematics
   std::vector<Eigen::Index> redundancy_indices_; /**< @brief Joint indicies that have redundancy (ex. revolute) */
   std::string name_;                             /**< @brief Name of the kinematic chain */
   std::string solver_name_;                      /**< @brief Name of this solver */
+  std::vector<double> free_joint_values_;        /**< @brief values to sample the free joint in case of a 7 DOF robot */
```

Comment:
I thought about this as well. In the MoveIt docs I [found](url) it says: 

> While it works in theory, currently the IKFast plugin generator tool does not work with >7 degree of freedom arms.

This could be outdated, but I am unsure. I am not sure exactly what the interface would look like if we are giving multiple values to the ComputeIk method, so that is why I restrained from doing what you suggested

Reply:
This has now been implemented to support >7 DOF